### PR TITLE
Fix Save button stop working issue

### PR DESCRIPTION
**Pull request by usingtechnology**
# Description
Fix the intended logic when adding a component with the API Key of `form`. Errors in that logic prevented the notification from displaying and prompting the form designer to remove that component.

See [FORMS-982](https://bcdevex.atlassian.net/browse/FORMS-982).
The issue reported was save button stops working. 

CHEFs did (intend to) have a prompt to notify the designer that they cannot save a component with API key = `form`. That's where the button gets disabled. Without the prompt the designer doesn't know how much work or how many components would be lost to reset the button by clicking `Undo`.

## Types of changes

Bug fix (non-breaking change which fixes an issue)

## Checklist

- [x] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [x] I have checked that unit tests pass locally with my changes
- [x] I have run the npm script lint on the frontend and backend
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)
- [x] I have approval from the product owner for the contribution in this pull request

## Further comments

Steps to reproduce (on existing code - use the dev instance)

1. Using Chrome, so you can open the Developer Tools | Javascript console.
2. Go to our dev instance and login with your IDIR - [CHEFS Dev](https://chefs-dev.apps.silver.devops.gov.bc.ca/app)
3. Create a new Form, the setup information doesn’t matter, I did Public, no API Key.
4. In design, click Save, make sure it works.
5. Add a Panel, just save the component, no need to rename it.
6. Save Form again, ensure button works.
7. Drag on a component, could be a panel, checkbox, text… doesn’t really matter. Set the API property name = form, save the component to close the dialog.
8. You should see an error stack in the Javascript console.
9. The Form Designer Save button should be grey and not respond to clicks.
10. Drag another component out, don’t bother renaming anything, save it to close the dialog
11. Save button still disabled.

On this PR, at step 7 you should see an error notification indicating you cannot have a component named `form`. The component is in the form, so remove it or  click `Undo`. You should then see the save button enabled and carry on adding/editing as before.

### CODE REVIEW

**Review by jasonchung1871 — APPROVED** [review]

File: app/frontend/src/components/designer/FormDesigner.vue [hunk 1]
```diff
@@ -431,11 +430,13 @@ export default {
...
-                ...NotificationTypes.ERROR,
-                message: this.$t('trans.formDesigner.fieldnameError', {
-                  label: comp.label,
-                }),
+              const notificationStore = useNotificationStore();
+              const msg = i18n.t('trans.formDesigner.fieldnameError', {
+                label: comp.label,
+              });
+              notificationStore.addNotification({
+                text: msg,
```
Comment (WalterMoar) [inline app/frontend/src/components/designer/FormDesigner.vue:438]: It looks like "message" is still being used in EmailTemplate and UserSubmission - if these also need to be updated, would it be better to put them in this PR, or to open new Jira item(s)?

Comment (usingtechnology) [reply app/frontend/src/components/designer/FormDesigner.vue:438]: let me find that and deal with it in this PR. Great find!

Comment (WalterMoar) [reply app/frontend/src/components/designer/FormDesigner.vue:438]: Looks great!

**Review by WalterMoar — APPROVED** [review]
Looks great!

**Pull request merged**